### PR TITLE
DynamoDB respect HTTP Connect proxies

### PR DESCRIPTION
**Pull request by kgtw**
This PR fixes (https://github.com/gravitational/teleport/issues/4261) allowing DynamoDB to make calls via a HTTP proxy.

### CODE REVIEW

**Comment by webvictim** [comment]
ok to test

**Comment by webvictim** [comment]
retest this please

File: lib/backend/dynamo/dynamodbbk.go [hunk 1]
```diff
@@ -217,14 +217,21 @@ func New(ctx context.Context, params backend.Params) (*DynamoDBBackend, error) {
...
 	// performance of Teleport under load as it reduces the number of TLS
 	// handshakes performed.
+	tr.MaxIdleConns = defaults.HTTPMaxIdleConns
+	tr.MaxIdleConnsPerHost = defaults.HTTPMaxConnsPerHost
+
 	httpClient := &http.Client{
-		Transport: &http.Transport{
-			MaxIdleConns:        defaults.HTTPMaxIdleConns,
-			MaxIdleConnsPerHost: defaults.HTTPMaxIdleConnsPerHost,
-		},
```
Comment (awly) [inline lib/backend/dynamo/dynamodbbk.go:227]: Set `Proxy: ProxyFromEnvironment,` here, instead of modifying `http.DefaultTransport`.

Comment (kgtw) [reply lib/backend/dynamo/dynamodbbk.go]: Updated with requested changes.

**Review by awly — COMMENTED** [review]
Please also format the code with `go fmt ./lib/backend/dynamo`

**Review by awly — APPROVED** [review]

**Review by webvictim — APPROVED** [review]

**Comment by kgtw** [comment]
Hi @awly, any chance of getting this merged and included in the next release? We would really like to move some of our secure environments over to using Teleport.

**Comment by webvictim** [comment]
@russjones What do you think?

**Comment by benarent** [comment]
@russjones This came up today, this would be good issue to backport into 4.3.6 https://github.com/gravitational/teleport/milestone/43

**Review by russjones — APPROVED** [review]

**Comment by webvictim** [comment]
Fixes #4261

**Pull request merged**